packages/wrangler/src/__tests__/r2.test.ts

```diff
@@ -2277,6 +2379,362 @@ describe("r2", () => {
 				});
 			});
 		});
+		describe("lock", () => {
+			const { setIsTTY } = useMockIsTTY();
+			mockAccountId();
+			mockApiToken();
+			describe("list", () => {
+				it("should list lock rules when they exist", async () => {
+					const bucketName = "my-bucket";
+					const lockRules = [
+						{
+							id: "rule-age",
+							enabled: true,
+							prefix: "images/age",
+							condition: {
+								type: "Age",
+								maxAgeSeconds: 86400,
+							},
+						},
+						{
+							id: "rule-date",
+							enabled: true,
+							prefix: "images/date",
+							condition: {
+								type: "Date",
+								date: 1738277955891,
+							},
+						},
+						{
+							id: "rule-indefinite",
+							enabled: true,
+							prefix: "images/indefinite",
+							condition: {
+								type: "Indefinite",
+							},
+						},
+					];
+					msw.use(
+						http.get(
+							"*/accounts/:accountId/r2/buckets/:bucketName/lock",
+							async ({ params }) => {
+								const { accountId, bucketName: bucketParam } = params;
+								expect(accountId).toEqual("some-account-id");
+								expect(bucketParam).toEqual(bucketName);
+								return HttpResponse.json(
+									createFetchResult({
+										rules: lockRules,
+									})
+								);
+							},
+							{ once: true }
+						)
+					);
+					await runWrangler(`r2 bucket lock list ${bucketName}`);
+					expect(std.out).toMatchInlineSnapshot(`
+					"Listing lock rules for bucket 'my-bucket'...
+					id:         rule-age
+					enabled:    Yes
+					prefix:     images/age
+					condition:  after 1 day
+
+					id:         rule-date
+					enabled:    Yes
+					prefix:     images/date
+					condition:  on 2025-01-30
+
+					id:         rule-indefinite
+					enabled:    Yes
+					prefix:     images/indefinite
+					condition:  indefinitely"
+				  `);
+				});
+			});
+			describe("add", () => {
+				it("it should add a lock rule without prefix using command-line arguments", async () => {
+					setIsTTY(false);
```

Comment:
```suggestion
					setIsTTY(true);
```
otherwise the prompts are just using the defaults